modules/build/src/main/scala/scala/build/options/BuildOptions.scala

```diff
@@ -292,9 +302,66 @@ final case class BuildOptions(
   def finalRepositories: Seq[String] =
     classPathOptions.extraRepositories ++ internal.localRepository.toSeq
 
-  private lazy val maxSupportedStableScalaVersions = latestSupportedStableScalaVersion()
+  private lazy val maxSupportedStableScalaVersions: Seq[Version] =
+    latestSupportedStableScalaVersion()
+
+  private lazy val latestSupportedStableVersions: Seq[String] =
+    maxSupportedStableScalaVersions.map(_.repr)
```

Comment:
Would it be possible _not_ to add types to existing code this way? This makes the diff harder to read, and doesn't really bring much in itself…

Reply:
Well, frankly, for myself, reading the code without marking with types is very difficult, because I then have to hover over something or look at the source of its implementation to know what the type is!